ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java

```diff
@@ -352,6 +353,82 @@ void check() throws Exception {
     opChain.testFutures.add(testFuture);
   }
 
+  /**
+   * Submit a getBlob operation with incorrect accountId/ContainerId in blobId..
+   * @param opChain the {@link OperationChain} object that this operation is a part of.
+   */
+  private void startGetBlobAuthorizationFailTest(final OperationChain opChain) {
+    Callback<GetBlobResult> callback = new TestCallback<>(opChain, false);
+    BlobId originalId, fraudId;
+    try {
+      originalId = new BlobId(opChain.blobId, clusterMap);
+      fraudId = BlobId.craft(originalId, originalId.getVersion(), (short) (originalId.getAccountId() + 1),
+          (short) (originalId.getContainerId() + 1));
+    } catch (IOException e) {
+      // If there is a blobId creation failure, throw the exception and don't need to do actual action.
+      FutureResult<Void> future = new FutureResult<>();
+      // continue the chain
+      future.done(null, null);
+      callback.onCompletion(null, null);
+      TestFuture<Void> testFuture = new TestFuture<Void>(future, genLabel("getBlob", true), opChain) {
+        @Override
+        void check() throws Exception {
+          throw e;
+        }
+      };
+      opChain.testFutures.add(testFuture);
+      return;
+    }
+    // If everything good:
+    Future<GetBlobResult> future = router.getBlob(fraudId.getID(),
+        new GetBlobOptionsBuilder().operationType(GetBlobOptions.OperationType.All)
+            .getOption(GetOption.Include_All)
+            .build(), callback);
+    TestFuture<GetBlobResult> testFuture = new TestFuture<GetBlobResult>(future, genLabel("getBlob", true), opChain) {
+      @Override
+      void check() throws Exception {
+        checkExpectedRouterErrorCode(RouterErrorCode.BlobAuthorizationFailure);
+      }
+    };
+    opChain.testFutures.add(testFuture);
+  }
+
+  /**
+   * Submit a deleteBlob operation with incorrect accountId/ContainerId in blobId.
+   * @param opChain the {@link OperationChain} object that this operation is a part of.
+   */
+  private void startDeleteBlobAuthorizationFailTest(final OperationChain opChain) {
```

Comment:
quick question: do we have blob authorization succeeded tests for `delete` and `get` operations?

Reply:
never mind. I notice that the normal `get` and `delete` will succeed, that's ok then.